components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -5,3 +5,597 @@
+		})
+
+		playerIndex = playerIndex + 1
+		prefix = teamName .. '_p' .. playerIndex
+		playerName = Variables.varDefault(prefix)
+	end
+
+	return players
+end
+
+--[[
+
+MapInput functions
+
+]]--
+function CustomMatchGroupInput._mapInput(match, mapIndex, subGroupIndex)
+	local map = Json.parseIfString(match['map' .. mapIndex]) or {}
+
+	if Table.isEmpty(map) then
+		match['map' .. mapIndex] = nil
+		return match, subGroupIndex
+	end
+
+	-- CR has no map names, use generic one instead
+	map.map = 'Set ' .. mapIndex
+
+	-- set initial extradata for maps
+	map.extradata = {
+		comment = map.comment,
+		header = map.header,
+	}
+
+	-- inherit stuff from match data
+	map.type = Logic.emptyOr(map.type, match.type)
+	map.liquipediatier = match.liquipediatier
+	map.liquipediatiertype = match.liquipediatiertype
+	map.game = Logic.emptyOr(map.game, match.game)
+	map.date = Logic.emptyOr(map.date, match.date)
+
+	-- determine score, resulttype, walkover and winner
+	map = CustomMatchGroupInput._mapWinnerProcessing(map)
+
+	-- get participants data for the map + get map mode
+	map = CustomMatchGroupInput._processPlayerMapData(map, match)
+
+	-- set sumscore to 0 if it isn't a number
+	if String.isEmpty(match.opponent1.sumscore) then
+		match.opponent1.sumscore = 0
+	end
+	if String.isEmpty(match.opponent2.sumscore) then
+		match.opponent2.sumscore = 0
+	end
+
+	--adjust sumscore for winner opponent
+	if (tonumber(map.winner) or 0) > 0 then
+		match['opponent' .. map.winner].sumscore =
+			match['opponent' .. map.winner].sumscore + 1
+	end
+
+	match['map' .. mapIndex] = map
+
+	return match, subGroupIndex
+end
+
+function CustomMatchGroupInput._mapWinnerProcessing(map)
+	map.scores = {}
+	local hasManualScores = false
+	local indexedScores = {}
+	for scoreIndex = 1, MAX_NUM_OPPONENTS do
+		-- read scores
+		local score = map['score' .. scoreIndex]
+		local obj = {}
+		if Logic.isNotEmpty(score) then
+			hasManualScores = true
+			score = CONVERT_STATUS_INPUT[score] or score
+			if Logic.isNumeric(score) then
+				obj.status = 'S'
+				obj.score = score
+			elseif Table.includes(ALLOWED_STATUSES, score) then
+				obj.status = score
+				obj.score = -1
+			end
+			table.insert(map.scores, score)
+			indexedScores[scoreIndex] = obj
+		else
+			break
+		end
+	end
+
+	if hasManualScores then
+		for scoreIndex, _ in Table.iter.spairs(indexedScores, CustomMatchGroupInput._placementSortFunction) do
+			if not tonumber(map.winner or '') then
+				map.winner = scoreIndex
+			else
+				break
+			end
+		end
+	else
+		local winnerInput = tonumber(map.winner)
+		if String.isNotEmpty(map.walkover) then
+			local walkoverInput = tonumber(map.walkover)
+			if walkoverInput == 1 then
+				map.winner = 1
+			elseif walkoverInput == 2 then
+				map.winner = 2
+			elseif walkoverInput == 0 then
+				map.winner = 0
+			end
+			map.walkover = Table.includes(ALLOWED_STATUSES, map.walkover) and map.walkover or 'L'
```

Comment:
winner is not set if walkover is a ALLOWED_STATUS input.
Also those branches can be made into one

Reply:
if branches merged

> winner is not set if walkover is a ALLOWED_STATUS input.

yes the idea is that if walkover is a number it does determine the winner
else it determines the walkover type and a manual winner input is expected